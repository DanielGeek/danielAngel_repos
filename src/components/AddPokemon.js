import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddPokemon = ({ setPokemones }) => {

  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (e) => {
    setInputValue(e.target.value.toLowerCase());
  }

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 0) {
      setPokemones( pokemones => {
        const newpokemones = pokemones.filter( pokemon => pokemon !== inputValue)
        return [ inputValue, ...newpokemones]
      });
      setInputValue('');
    }

  }

  return (
    <>
      <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
        <input
          type="text"
          placeholder='Buscar'
          value={inputValue}
          onChange={handleInputChange}
        />
      </form>
    </>
  )
}

AddPokemon.propTypes = {
  setPokemones: PropTypes.func.isRequired
}
