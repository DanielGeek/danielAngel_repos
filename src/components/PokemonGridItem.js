import React from 'react';
import PropTypes from 'prop-types';
import { capitalizeFirstLetter } from '../helpers/capitalizeFirstLette';

export const PokemonGridItem = ({ name, url, id }) => {

  return (
    <div className="card animate__animated animate__fadeIn">
      <img src={ url } alt={ name } />
      <p># { id }</p>
      <p>{ capitalizeFirstLetter(name) }</p>
    </div>
  )
}

PokemonGridItem.propTypes = {
  name: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  id: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
}
