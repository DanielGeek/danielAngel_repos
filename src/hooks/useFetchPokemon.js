import { useEffect, useState } from "react"
import { getPokemon } from "../helpers/getPokemon";

export const useFetchPokemon = ( pokemonName ) => {
  const [state, setState] = useState({
    data: [],
    loading: true,
    error: false,
  });

  useEffect(() => {

    getPokemon( pokemonName )
      .then( imgs => {

        setState({
          data: imgs,
          loading: false,
          error: false,
        });

      }).catch(error => {
        setState({
          data: [],
          loading: false,
          error:error
        });
      });


  }, [pokemonName])


  return state;
}