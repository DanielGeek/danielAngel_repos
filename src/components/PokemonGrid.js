import React from 'react'
import PropTypes from 'prop-types';

import { useFetchPokemon } from '../hooks/useFetchPokemon'
import { PokemonGridItem } from './PokemonGridItem';

export const PokemonGrid = ({ pokemonNameOrId }) => {

  const { data: images, loading, error } = useFetchPokemon( pokemonNameOrId );

  let imagesFilters = images.filter(image => image.id !== pokemonNameOrId);

  return (
    <>

      { loading && <p className='animate__animated animate__flash'>Cargando</p>}
      { !error &&
          <div className='card-grid'>
          {
            imagesFilters.map( img => (
              <PokemonGridItem
                  key={ img.id }
                  { ...img }
              />
            ))
          }
        </div>
      }
    </>
  )
}

PokemonGrid.propTypes = {
  pokemonNameOrId: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
}