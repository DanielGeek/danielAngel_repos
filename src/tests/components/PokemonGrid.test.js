import React from 'react';
import '@testing-library/jest-dom';

import {PokemonGrid} from '../../components/PokemonGrid';
import { shallow } from 'enzyme';
import { useFetchPokemon } from '../../hooks/useFetchPokemon';
jest.mock('../../hooks/useFetchPokemon');

describe('Tests for PokemonGrid component', () => {
    const pokemonNameOrId = 'Bulbasaur';
    // const wrapper = shallow(<PokemonGrid pokemonNameOrId={pokemonNameOrId}/>);

    test('debe de mostrarse correctamente', () => {

        useFetchPokemon.mockReturnValue({
            data: [],
            loading: true,
        });
        const wrapper = shallow(<PokemonGrid pokemonNameOrId={pokemonNameOrId}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imÃ¡genes useFetchPokemon', () => {

        const pokemon = [{
            id: '1',
            url: 'https://google.com',
            name: 'Bulbasaur'
        }]

        useFetchPokemon.mockReturnValue({
            data: pokemon,
            loading: false,
            error: false,
        });

        const wrapper = shallow(<PokemonGrid pokemonNameOrId={pokemonNameOrId}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('PokemonGridItem').length).toBe(pokemon.length);


    });
})