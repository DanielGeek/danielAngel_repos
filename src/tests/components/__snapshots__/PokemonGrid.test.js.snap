// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for PokemonGrid component debe de mostrar items cuando se cargan im√°genes useFetchPokemon 1`] = `
<Fragment>
  <div
    className="card-grid"
  >
    <PokemonGridItem
      id="1"
      key="1"
      name="Bulbasaur"
      url="https://google.com"
    />
  </div>
</Fragment>
`;

exports[`Tests for PokemonGrid component debe de mostrarse correctamente 1`] = `
<Fragment>
  <p
    className="animate__animated animate__flash"
  >
    Cargando
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
